### PR TITLE
vim-patch:partial:6ebe4f970b8b

### PR DESCRIPTION
**Pull request by clason**
Update runtime files

https://github.com/vim/vim/commit/6ebe4f970b8b398087076a72a7aae6e680fb92da

Co-authored-by: Bram Moolenaar <Bram@vim.org>

* Needs `'endoffile'` patch
* Also revert `syntax/vim.vim` change from f269eabc6c4f
* Update URL in `tui.c` instead of docs

### CODE REVIEW

**Comment by zeertzjq** [comment]
Missing some space -> tab changes in ft_rust.txt

File: runtime/doc/options.txt [hunk 1]
```diff
@@ -2139,7 +2139,7 @@ A jump table for the options with a short description can be found at |Q_op|.
 'endofline' 'eol'	boolean	(default on)
 			local to buffer
 	When writing a file and this option is off and the 'binary' option
-	is on, or 'fixeol' option is off, no <EOL> will be written for the
+	is on, or 'fixeol' option is off, no <EOL> will be written at the
```
Comment (zeertzjq) [inline runtime/doc/options.txt:2142]: keep

**Comment by clason** [comment]
Which ones? They're impossible to see from the diff.

(Can't we just retab the whole files and be done with it, once and for all?)

File: runtime/doc/options.txt (hunk 1 above)
Comment (clason) [reply runtime/doc/options.txt:2142]: Why?

Comment (zeertzjq) [reply runtime/doc/options.txt:2142]: Because Vim has not changed this.

Comment (zeertzjq) [reply runtime/doc/options.txt:2142]: See https://github.com/neovim/neovim/pull/20855/files

Comment (clason) [reply runtime/doc/options.txt:2142]: Sorry, I don't see the point. There was a typo (`for at`), which we fixed one way at some point and Bram fixed another way. We tend to follow upstream unless there are pressing reasons, so I switched the fix.

File: runtime/doc/scroll.txt [hunk 2]
```diff
@@ -251,4 +251,12 @@ scrolled. This allows you to scroll inactive windows. Note that when scroll
 keys are remapped to keyboard keys, the active window is affected regardless
 of the current cursor position.
 
+							*mouse-scrolling-off*
+If you do not want the mouse to cause scrolling (e.g. because resting your
+palm on the touchpad causes scroll events), you can disable that with: >
+	:map <ScrollWheelDown> <Nop>
+	:map! <ScrollWheelDown> <Nop>
+	:map <ScrollWheelUp> <Nop>
+	:map! <ScrollWheelUp> <Nop>
```
Comment (seandewar) [inline runtime/doc/scroll.txt:260]: Nvim technically suggests `:set mousescroll=hor:0,ver:0` in `:h 'mousescroll'` to disable mouse scrolling, but it seems to still move the cursor. This mapping imo works a bit better as it doesn't do that. :shrug:

File: runtime/doc/options.txt (hunk 1 above)
Comment (zeertzjq) [reply runtime/doc/options.txt:2142]: Why is there a typo?

File: runtime/doc/ft_rust.txt [hunk 3]
```diff
@@ -1,70 +1,70 @@
-*ft_rust.txt*	For Vim version 8.1.  Last change: 2017 Nov 02
+*ft_rust.txt*	For Vim version 9.0.  Last change: 2022 Oct 17
```
Comment (seandewar) [inline runtime/doc/ft_rust.txt:1]:
```suggestion
*ft_rust.txt*	Nvim
```

**Review by seandewar — COMMENTED** [review]
Could cherry-pick this tab at: https://github.com/vim/vim/commit/6ebe4f970b8b398087076a72a7aae6e680fb92da#diff-e891ea60176f3f0bf0aa9bcd13c22cdb77f377ea932fe9515f5e0b16df8740daR5299

File: runtime/doc/options.txt (hunk 1 above)
Comment (clason) [reply runtime/doc/options.txt:2142]: Oh, I see what you mean, I was looking at the wrong section.

(Using more than a single word for comments may safe some tedious -- for both parties -- back and forth.)

Comment (zeertzjq) [reply runtime/doc/options.txt:2142]: The current docs is exactly the same as upstream, and you are diverging from upstream by changing this.

File: runtime/doc/ft_rust.txt [hunk 4]
```diff
@@ -76,63 +76,63 @@ g:rust_fold~
...
 	|location-list| with errors. If not specified it defaults to 0: >
 	    let g:rustfmt_fail_silently = 0
 <
-                                                       *g:rustfmt_options*
+						       *g:rustfmt_options*
 g:rustfmt_options~
-	Set this option to a string of options to pass to "rustfmt". The
-	write-mode is already set to "overwrite". If not specified it
+	Set this option to a string of options to pass to 'rustfmt'. The
+	write-mode is already set to 'overwrite'. If not specified it
```
Comment (seandewar) [inline runtime/doc/ft_rust.txt:114]: Justin changed these to ", I think.

File: runtime/doc/scroll.txt (hunk 2 above)
Comment (zeertzjq) [reply runtime/doc/scroll.txt:260]: Does it?

Comment (zeertzjq) [reply runtime/doc/scroll.txt:260]: And why aren't left and right scrolling keys mapped here?

Comment (seandewar) [reply runtime/doc/scroll.txt:260]: > Does it?

`:h 'mousescroll'` mentions "You can disable mouse scrolling by using a count of 0.". While this disables scrolling, the cursor still seems to move for me (with the mapping it doesn't). :thinking:

Comment (seandewar) [reply runtime/doc/scroll.txt:260]: Funnily enough I can even move it outside of the window's view, and the window doesn't scroll until I move the cursor some other way :thinking:

Comment (seandewar) [reply runtime/doc/scroll.txt]: (Should be fixed by #20861)

Comment (clason) [reply runtime/doc/scroll.txt]: It's gone now, anyway.

**Review by seandewar — APPROVED** [review]

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**